Kernel/Devices/HID/PS2/KeyboardDevice.cpp

```diff
@@ -19,28 +19,693 @@
+            key_event.scancode = 0xf000 + byte;
+        } else {
+            VERIFY_NOT_REACHED();
+        }
+        m_entropy_source.add_random_event(byte);
+    } else if (event.bytes_count == 3) {
+        auto first_byte_prefix = event.scan_code_bytes[0];
+        auto second_byte_prefix = event.scan_code_bytes[1];
+        VERIFY(first_byte_prefix == 0xe0);
+        VERIFY(second_byte_prefix == 0xf0);
+        auto byte = event.scan_code_bytes[2];
+        key_event.code_entry = unshifted_scan_code_set2_e0_key_map[byte];
+        key_event.scancode = 0xe0f000 + byte;
+        m_entropy_source.add_random_event(byte);
+    } else if (event.bytes_count == 4) {
+        // 0xE0, 0x12, 0xE0, 0x7C - print screen pressed
+        auto first_byte_prefix = event.scan_code_bytes[0];
+        auto second_byte_prefix = event.scan_code_bytes[1];
+        auto third_byte_prefix = event.scan_code_bytes[2];
+        auto fourth_byte_prefix = event.scan_code_bytes[3];
+        VERIFY(first_byte_prefix == 0xe0);
+        VERIFY(second_byte_prefix == 0x12);
+        VERIFY(third_byte_prefix == 0xe0);
+        VERIFY(fourth_byte_prefix == 0x7c);
+        key_event.code_entry = KeyCodeEntry { Key_PrintScreen, 0xFF };
+        key_event.scancode = 0xe012e07c;
+        key_event.is_press_down = true;
+    } else if (event.bytes_count == 6) {
+        // 0xE0, 0xF0, 0x7C, 0xE0, 0xF0, 0x12 - print screen released
+        auto first_byte_prefix = event.scan_code_bytes[0];
+        auto second_byte_prefix = event.scan_code_bytes[1];
+        auto third_byte_prefix = event.scan_code_bytes[2];
+        auto fourth_byte_prefix = event.scan_code_bytes[3];
+        auto fifth_byte_prefix = event.scan_code_bytes[4];
+        auto sixth_byte_prefix = event.scan_code_bytes[5];
+        VERIFY(first_byte_prefix == 0xe0);
+        VERIFY(second_byte_prefix == 0xf0);
+        VERIFY(third_byte_prefix == 0x7c);
+        VERIFY(fourth_byte_prefix == 0xe0);
+        VERIFY(fifth_byte_prefix == 0xf0);
+        VERIFY(sixth_byte_prefix == 0x12);
+        key_event.code_entry = KeyCodeEntry { Key_PrintScreen, 0xFF };
+        key_event.scancode = 0xe0f07ce0f012;
+    } else if (event.bytes_count == 8) {
+        // 0xE1, 0x14, 0x77, 0xE1, 0xF0, 0x14, 0xF0, 0x77 - pause pressed
+        auto first_byte_prefix = event.scan_code_bytes[0];
+        auto second_byte_prefix = event.scan_code_bytes[1];
+        auto third_byte_prefix = event.scan_code_bytes[2];
+        auto fourth_byte_prefix = event.scan_code_bytes[3];
+        auto fifth_byte_prefix = event.scan_code_bytes[4];
+        auto sixth_byte_prefix = event.scan_code_bytes[5];
+        auto seventh_byte_prefix = event.scan_code_bytes[6];
+        auto eight_byte_prefix = event.scan_code_bytes[7];
+        VERIFY(first_byte_prefix == 0xe1);
+        VERIFY(second_byte_prefix == 0x14);
+        VERIFY(third_byte_prefix == 0x77);
+        VERIFY(fourth_byte_prefix == 0xe1);
```

Comment:
In a follow-up, can we just drop the keyboard event if it's an unexpected multi-byte sequence, with a complaint in the log? If I wave a magnet over the cable while typing pause-break, we shouldn't crash the kernel.

Reply:
Should we do it in this PR? or a future commit?

Reply:
future